### ISSUE TITLE
Autojoin for integrated raft storage and SSL issues

### ISSUE THREAD

**Issue by awolde** [hashicorp/vault#9486]
In my `vault.hcl` config I have to list all the "expected" IPs of vault nodes like this to setup integrated storage:

```
storage "raft" {
  path    = "/opt/vault/"
  node_id = "node-${INSTANCE_ID}"
  retry_join {
    leader_api_addr = "https://${IP1}:8200"
  }
  retry_join {
    leader_api_addr = "https://${IP2}:8200"
  }
}
```

**Describe the solution you'd like**
Having something like what Consul is doing with autojoin will greatly help:
```
retry_join          = ["provider=gce project_name=${project} tag_value=my-consul"]
```

Also if I am using a single SSL cert for my Vault cluster, the `retry_join` block wont work because there is no way I can map the cert's CN to different IPs. The only way I can use TLS is if I have some dynamic way of DNS updates when nodes in the cluster come and go and a wildcard cert for all my nodes. Maybe leverage the 8201 port for raft backend??

**Describe alternatives you've considered**
Because of the above reasons, I'm still using Consul backend.

**Comment by yermulnik** [comment]
For anyone popping in to this issue from google search: the auto join feature has been implemented in vault v1.6.0 released the other day.

**Comment by mladlow** [comment]
As @yermulnik pointed out, we did add auto-join in the 1.6 Vault release, so I'm going to close this issue.
